scripts/run_benchmark.py

```diff
@@ -300,6 +301,9 @@ def init_worker(env: Dict) -> NoReturn:
     )
 
     args = parser.parse_args()
+    if args.gpus[0] == -1:
+        print("Using All GPUs Detected.")
+        args.gpus = np.arange(torch.cuda.device_count()).tolist()
```

Comment:
Using `-1` to represent all GPUs might cause problems in rare cases: because the minus sign is the same as the hyphen prefixes of command line arguments, there might be some naive parser (for example, highlighting in certain IDEs) that mistakenly intepret '-1' as an argument. So I suggest to make the following changes:

- change the gpus from `type=int` to `type=str`
- add parse functions for this argument, allowing inputs like `--gpus all`, `--gpus 0 1 2`, `--gpus 0,1,2`, the implementation should be something like:
```python
def parse_gpus(gpu_list: List[str]) -> List[int]:
    """docstring here
    """
    gpu_ids = set()

    for item in gpu_list:
        if item.strip().lower() == "all":
            gpu_ids.update(get_all_gpus())
        else:
            try:
                # Split by comma and convert to integers
                ids = map(int, item.split(','))
            except ValueError as e:
                raise ValueError(f"Invalid GPU ID in input '{item}': {e}")

            if any(it < 0 for it in ids):
                raise ValueError(f"GPU ID cannot be negative.")

            gpu_ids.update(ids)
    return sorted(gpu_ids)
```

Reply:
Given that I remain uncertain as to whether modifying the type of `args.gpus` would lead to any issues within other sections of TFB, and since there has been no consensus reached regarding the correct interpretation of this matter, I have opted to retain the current design at this stage.